File: HotelWeb/src/app/models/hotel.ts
export interface Hotel {
  id: string;
  naziv: string;
  kontaktBroj: string;
  kontaktEmail: string;
  adresa: string;
  urlSlike: string;
  mjestoId?: string;
  mjestoNaziv?: string;
  statusHotelaId?: number;
  statusHotela?: string;
  upraviteljIds?: number[];
}
export interface UpsertHotel {
  naziv: string;
  kontaktBroj: string;
  kontaktEmail: string;
  adresa: string;
  urlSlike?: File;
  mjestoId?: string;
  statusHotelaId?: number;
  upraviteljIds?: number[];
}
export interface HotelStatus {
  id: number;
  naziv: string;
}

export interface HotelFilter {
  mjestoId?: string;
  statusHotelaId?: number;
  naziv?: string;
}

File: HotelAPI/Models/Entities/Hotel.cs
namespace HotelAPI.Models.Entities
{
    public class Hotel : Entity
    {
        public required string Naziv { get; set; }
        public required string KontaktBroj { get; set; }
        public required string KontaktEmail { get; set; }
        public required string Adresa { get; set; }
        public required string UrlSlike { get; set; }
        public Guid? MjestoId { get; set; }
        public Mjesto? Mjesto { get; set; }
        public int? StatusHotelaId { get; set; }
        public StatusHotela? StatusHotela { get; set; }
        public ICollection<Korisnik>? Upravitelji { get; set; } = new List<Korisnik>();
        public ICollection<Soba>? Sobe { get; set; }
        public bool IsDeleted { get; set; } = false;

    }
}
